test/e2e/ganache.js

```diff
@@ -11,12 +11,32 @@ const defaultOptions = {
   quiet: true,
 };
 
+// This is a "static" variable that holds the default Ganache instance
+let ganacheInstanceOnDefaultPort = null;
```

Comment:
I thought about that, but it can't be a true singleton because _**sometimes**_ there are two Ganache servers.  Search for the word `concurrent` using "Match Whole Word" and you'll see it.  That's why I did that whole checking for default port stuff.

Reply:
I made a bunch of changes based on your suggestions.  I think it's a lot better, but actually still not perfect (I still see it crash).  But let me know what you think you the changes so far.